ENH: TubeTK should support 2D images by default - used in tests

aylward opened the pull request:
ENH: Changed 2D support to default to ON.  This will also fix several failing tests.

STYLE: Fixed spacing issues in CMakeLists.txt file (needed spaces between parens and their content.

fbudin69500 commented:
Should these tests be disabled when compiling with 2D OFF since it would be
the expected behavior?

On Sat, Feb 27, 2016, 17:50 Stephen R. Aylward <notifications@github.com>
wrote:

> ENH: Changed 2D support to default to ON. This will also fix several
> failing tests.
>
> STYLE: Fixed spacing issues in CMakeLists.txt file (needed spaces between
> parens and their content.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/KitwareMedical/TubeTK/pull/618
> Commit Summary
>
>    - ENH: TubeTK should support 2D images by default - used in tests
>
> File Changes
>
>    - *M* CMakeLists.txt
>    <https://github.com/KitwareMedical/TubeTK/pull/618/files#diff-0> (63)
>
> Patch Links:
>
>    - https://github.com/KitwareMedical/TubeTK/pull/618.patch
>    - https://github.com/KitwareMedical/TubeTK/pull/618.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/TubeTK/pull/618>.
>

aylward commented:
How about requiring 2D turned on if testing is turned on?

s

On Mon, Feb 29, 2016 at 9:56 AM, Francois Budin <notifications@github.com>
wrote:

> Should these tests be disabled when compiling with 2D OFF since it would be
> the expected behavior?
>
> On Sat, Feb 27, 2016, 17:50 Stephen R. Aylward <notifications@github.com>
> wrote:
>
> > ENH: Changed 2D support to default to ON. This will also fix several
> > failing tests.
> >
> > STYLE: Fixed spacing issues in CMakeLists.txt file (needed spaces between
> > parens and their content.
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/KitwareMedical/TubeTK/pull/618
> > Commit Summary
> >
> > - ENH: TubeTK should support 2D images by default - used in tests
> >
> > File Changes
> >
> > - *M* CMakeLists.txt
> > <https://github.com/KitwareMedical/TubeTK/pull/618/files#diff-0> (63)
> >
> > Patch Links:
> >
> > - https://github.com/KitwareMedical/TubeTK/pull/618.patch
> > - https://github.com/KitwareMedical/TubeTK/pull/618.diff
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/KitwareMedical/TubeTK/pull/618>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/TubeTK/pull/618#issuecomment-190245967>
> .
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================

fbudin69500 commented:
@jcfr , what do you think it the cleanest? turning OFF the tests that fail when 2D is deactivated, or requiring 2D to be turned ON when the TESTING is ON?

aylward commented:
Hi

My thoughts...

Coverage would be very low if we only ran the 3d tests.

As an itk module, we need to test 2d and 3d.

We should probably make turning 2d support off be an advanced option.  It
should not be recommended.

S
On Feb 29, 2016 8:27 AM, "Francois Budin" <notifications@github.com> wrote:

> @jcfr <https://github.com/jcfr> , what do you think it the cleanest?
> turning OFF the tests that fail when 2D is deactivated, or requiring 2D to
> be turned ON when the TESTING is ON?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/TubeTK/pull/618#issuecomment-190276850>
> .
>

jcfr commented:
Independently of your choice, just make sure that cases with `TubeTK_Support_2D_Images` disabled and enabled are tested.

Ideally, as a user, if I call a program that could be compiled with option either enabled or disabled, appropriate error message and outputs should be reported.

aylward commented:
Hi,

If testing is on, let's require 2D to be on.   I will make this change and
submit a pull request as part of test fixes I am generating now.

s

On Mon, Feb 29, 2016 at 1:31 PM, Jean-Christophe Fillion-Robin <
notifications@github.com> wrote:

> Independently of your choice, just make sure that cases with
> TubeTK_Support_2D_Images disabled and enabled are tested.
>
> Ideally, as a user, if I call a program that could be compiled with option
> either enabled or disabled, appropriate error message and outputs should be
> reported.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/TubeTK/pull/618#issuecomment-190323158>
> .
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================

fbudin69500 commented:
That sounds good since adding 2D support doesn't break anything.
